File: backend/src/main/java/org/sport/backend/service/PaymentService.java
package org.sport.backend.service;

import org.sport.backend.constant.PaymentMethod;
import org.sport.backend.dto.base.PageResponse;
import org.sport.backend.dto.request.payment.CheckoutRequest;
import org.sport.backend.dto.response.payment.CheckoutResponse;
import org.sport.backend.dto.response.payment.RefundResponse;
import org.springframework.transaction.annotation.Transactional;

import java.util.Map;
import java.util.UUID;

public interface PaymentService {
    CheckoutResponse checkout(CheckoutRequest checkoutRequest);

    CheckoutResponse checkoutPayment(CheckoutRequest checkoutRequest);

    Map<String, Object> handlePayOsWebhook(Map<String, Object> payload);

    CheckoutResponse handleCheckoutResult(String orderCode, String status);

    CheckoutResponse handleVnPayReturn(Map<String, String> queryParams);

    @Transactional
    CheckoutResponse checkoutMatchJoin(UUID registrationId, PaymentMethod method);

    PageResponse<RefundResponse> getPendingRefunds(int page, int size);

    PageResponse<RefundResponse> getCompletedRefunds(int page, int size);

    @Transactional
    void confirmManualRefund(UUID paymentId);
}

File: frontend/src/service/payment/paymentService.ts
import api from "../../config/axios";

class PaymentService {
  async checkoutPayment(
    bookingIntentId: string,
    paymentMethod: string,
    isDeposit: boolean,
  ) {
    const res = await api.post("/payments/checkout-payment", {
      bookingIntentId,
      paymentMethod,
      isDeposit: isDeposit,
    });
    return res.data;
  }

  async checkoutMatchJoin(registrationId: string, paymentMethod: string) {
    const res = await api.post("/payments/checkout-match", {
      registrationId,
      paymentMethod,
    });
    return res.data;
  }

  async handleBookingPaymentResult(data: {
    orderCode: string;
    status: string;
  }) {
    return await api.get("/payments/result", {
      params: data,
    });
  }

  async handleVnPayReturn(queryString: string) {
    return await api.get(`/payments/vnpay/return${queryString}`);
  }
}

export default new PaymentService();
